### PR TITLE
upload MalwareBazaar

### PR DESCRIPTION
**Pull request by eyalpalo**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/40594

## Description
MalwareBazaar integration

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar.yml [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+      type: String
+    - contextPath: MalwareBazaar.File.imphash
+      description: imphash (only available for PE executables).
+      type: String
+    - contextPath: MalwareBazaar.File.tlsh
+      description: Trend Micro Locality Sensitive Hash (tlsh).
+      type: String
+    - contextPath: MalwareBazaar.File.telfhash
+      description: Trend Micro ELF Hash (telfhash).
+      type: Unknown
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar.yml:94]: please go over the "unknown" types and if strings, update accordingly

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar.yml [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+      type: String
+  - arguments:
+    - default: false
+      description: SHA256 hash of the malware sample to download.
+      isArray: false
+      name: sha256_hash
+      required: true
+      secret: false
+    deprecated: false
+    description: Download (fetch) malware samples from MalwareBazaar. Any malware
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar.yml:229]:
```suggestion
    description: Download malware samples from MalwareBazaar. Any malware
```
fetch is a "reserved word" in XSOAR, we use it only in fetch incidents/indicators

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar.yml [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+      type: String
+    - contextPath: File.Type
+      description: The file type.
+      type: String
+    - contextPath: File.MD5
+      description: The MD5 hash of the file.
+      type: String
+    - contextPath: File.Extension
+      description: The extension of the file.
+      type: String
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar.yml:267]:
```suggestion
```
`File` context path is for `file` commands

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_description.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+## MalwareBazaar
+
+MalwareBazaar offers API to download malware samples, comment malware samples and 
+obtain intel based of file hash, tag, signature, file type, etc.
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_description.md:4]:
```suggestion
MalwareBazaar offers API to download malware samples, comment malware samples and obtain intel based of file hash, tag, signature, file type, etc.
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
+import pytest
+
+import MalwareBazaar
+import CommonServerPython
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py:4]:
```suggestion
from CommonServerPython import Common
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    When:
+        - Running a file command
+
+    Then:
+        - creating a CommanResult containing DbotScore and relationship
+    """
+    mock_response = util_load_json('test_data/response_data_dbot_score.json')
+    result = MalwareBazaar.file_process("094fd325049b8a9cf6d3e5ef2a6d4cc6a567d7d49c35f8bb8dd9e3c6acf3d78d",
+                                        {}, mock_response)
+    indicator: CommonServerPython.Common.File = result.indicator
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py:49]:
```suggestion
    indicator: Common.File = result.indicator
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+    """
+
+    Given:
+        - response from MalwareBazaar on hash file
+
+    When:
+        - Running a file command
+
+    Then:
+        - creating a CommanResult containing DbotScore and relationship
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py:44]:
```suggestion
        - Make sure a CommandResult containing DbotScore with a BAD score and a relationship is returned.
```
also let's check the relationship

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+    """
+
+        Given:
+            - Request file reputation given hash array
+
+        When:
+            - Running a file reputation command
+
+        Then:
+            - Return file reputation for each file
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py:63]:
```suggestion
            - Make sure a file reputation for each file is returned.
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+        Given:
+            - sample_type to search by and sample_value to search for, limit and pagination is
+            optional
+
+        When:
+            - Running a list command to retreive a list of malware samples
+
+        Then:
+            - return a list of malware samples
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py:105]:
```suggestion
            - Make sure a list of malware samples is returned.
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+    """
+
+        Given:
+            - Hash of file and comment to add to MalwareBazaar db about this file
+
+        When:
+            - Running a comment add command
+
+        Then:
+            - Display a success message or raise exception when it fails
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py:129]:
```suggestion
            - Make sure a success message is returned.
```
also let's check the case it fails when no API key is provided.

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+    """
+
+    Given:
+        - Request with wrong arguments
+
+    When:
+        - Running a command
+
+    Then:
+        - Raising an error
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/MalwareBazaar_test.py:209]:
```suggestion
        - Make sure an error is raised.
```

File: Packs/MalwareBazaar/Integrations/MalwareBazaar/README.md [hunk 12]
```diff
@@ -0,0 +1,751 @@
+MalwareBazaar is a project from abuse.ch with the goal of sharing malware samples with the infosec community, AV vendors and threat intelligence providers.
+
```
Comment (dantavori) [inline Packs/MalwareBazaar/Integrations/MalwareBazaar/README.md:2]: why did you remove the line specifies the product version?

**Review by dantavori — CHANGES_REQUESTED** [review]
great job!
1. Test playbook failed to me on malware-comment-add's `Verify Outputs` task - please review
2. getting this when I try to run a command from my localhost (even though "trust any certificate" is checked):
   image
3. Go over the validation errors in the build.

see my other notes, and please schedule a demo (including me, Anar, Guy, Shahaf and Meital).

**Comment by ShirleyDenkberg** [comment]
@eyalpalo I edited the yml file.  Therefore, please regenerate the integration README.  I also added content for the pack README.  
@dantavori Doc review completed.

**Pull request merged**